modules/hostap/src/supp_api.c

```diff
@@ -599,6 +599,32 @@ static int wpas_add_and_config_network(struct wpa_supplicant *wpa_s,
 					goto out;
 				}
 			}
+		} else if (params->security == WIFI_SECURITY_TYPE_WPA_AUTO_PERSONAL) {
+			if (!wpa_cli_cmd_v("set_network %d psk \"%s\"", resp.network_id,
+					   psk_null_terminated)) {
+				goto out;
+			}
+
+			if (params->sae_password) {
+				if (!wpa_cli_cmd_v("set_network %d sae_password \"%s\"",
+						   resp.network_id, sae_null_terminated)) {
+					goto out;
+				}
+			} else {
+				if (!wpa_cli_cmd_v("set_network %d sae_password \"%s\"",
+						   resp.network_id, psk_null_terminated)) {
+					goto out;
+				}
+			}
+
+			if (!wpa_cli_cmd_v("set sae_pwe 2")) {
+				goto out;
+			}
+
+			if (!wpa_cli_cmd_v("set_network %d key_mgmt WPA-PSK SAE",
```

Comment:
shouldn't we limit to WPA2 like we do for `hapd` below?

Reply:
For `wpa_supplicant`, I think it defaults to WPA2 because it limits the protocol to RSN in line 546. 
If we change proto to WPA, it will be WPA-PSK.